chain/messagepool/provider.go

```diff
@@ -39,10 +42,19 @@ type Provider interface {
 	IsLite() bool
 }
 
+type nonceCacheKey struct {
+	types.TipSetKey
+	address.Address
+}
+
+var nonceCacheSize = 128
+
 type mpoolProvider struct {
 	sm *stmgr.StateManager
 	ps *pubsub.PubSub
 
+	liteActorCache *lru.Cache[nonceCacheKey, result.Result[*types.Actor]]
```

Comment:
Caching results also caches errors, and that is precisely the point - in lite mode if the node hasn't run for a while the head it is requesting may be behind lotus-gateway lookback limit, so every request would fail in the same way.